agents/mongodb/internal/profiler/collector/collector.go

```diff
@@ -223,73 +179,56 @@ func connectAndCollect(client *mongo.Client, dbName string, docsChan chan<- prot
 	// do not cancel cursor closing when ctx is canceled
 	defer cursor.Close(context.Background()) //nolint:errcheck
 
-	stats.IteratorCreated = time.Now().UTC().Format("2006-01-02 15:04:05")
-	stats.IteratorCounter += 1
-
 	// we got iterator, we are ready
 	signalReady(ready)
 
-	for {
+	// check if we should shutdown
+	select {
+	case <-doneChan:
+		return
+	default:
+		// just continue if not
+	}
+	for cursor.TryNext(timeoutCtx) {
```

Comment:
> This is only recommended for use with tailable cursors as a non-blocking alternative to Next.

This PR also switches cursor to non-tailable. It is also not clear why we want to avoid blocking.
And it is not easy to tell if it is working without a new test.

Reply:
Agree. Cannot see what's the benefit of making the cursor not blocking.

Reply:
It's the main reason why we got high CPU.

Reply:
Changes with startTime and endTime was done to not lose any data

Reply:
@AlekSi since we send data to qan once per minute my changes shouldn't affect at all.

Reply:
in the other hand previous code could lose some data because of gaps between requests.

Reply:
> It's the main reason why we got high CPU.

Blocking (doing nothing) can't cause high CPU usage. [Busy loop](https://en.wikipedia.org/wiki/Busy_waiting) can.

> in the other hand previous code could lose some data because of gaps between requests.

There must be a test showing that – it should fail with old code, and pass with a new one.

Reply:
I am going to write the test